core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -82,337 +83,10 @@ KOKKOS_INLINE_FUNCTION constexpr typename std::remove_reference<T>::type&& move(
-
-//----------------------------------------
-
-template <typename IntegerSequence, typename Result,
-          typename ResultSequence =
-              integer_sequence<typename IntegerSequence::value_type>>
-struct exclusive_scan_integer_sequence_helper;
-
-template <typename T, T h0, T... tail, typename Result, T... results>
-struct exclusive_scan_integer_sequence_helper<
-    integer_sequence<T, h0, tail...>, Result, integer_sequence<T, results...>>
-    : public exclusive_scan_integer_sequence_helper<
-          integer_sequence<T, tail...>,
-          std::integral_constant<T, Result::value + h0>,
-          integer_sequence<T, 0, (results + h0)...>> {};
-
-template <typename T, typename Result, T... results>
-struct exclusive_scan_integer_sequence_helper<integer_sequence<T>, Result,
-                                              integer_sequence<T, results...>> {
-  using type               = integer_sequence<T, results...>;
-  static constexpr T value = Result::value;
-};
-
-template <typename IntegerSequence>
-struct exclusive_scan_integer_sequence {
-  using value_type = typename IntegerSequence::value_type;
-  using helper     = exclusive_scan_integer_sequence_helper<
-      reverse_integer_sequence<IntegerSequence>,
-      std::integral_constant<value_type, 0>>;
-  using type                        = typename helper::type;
-  static constexpr value_type value = helper::value;
-};
-
-//----------------------------------------
-
-template <typename IntegerSequence, typename Result,
-          typename ResultSequence =
-              integer_sequence<typename IntegerSequence::value_type>>
-struct inclusive_scan_integer_sequence_helper;
-
-template <typename T, T h0, T... tail, typename Result, T... results>
-struct inclusive_scan_integer_sequence_helper<
-    integer_sequence<T, h0, tail...>, Result, integer_sequence<T, results...>>
-    : public inclusive_scan_integer_sequence_helper<
-          integer_sequence<T, tail...>,
-          std::integral_constant<T, Result::value + h0>,
-          integer_sequence<T, h0, (results + h0)...>> {};
-
-template <typename T, typename Result, T... results>
-struct inclusive_scan_integer_sequence_helper<integer_sequence<T>, Result,
-                                              integer_sequence<T, results...>> {
-  using type               = integer_sequence<T, results...>;
-  static constexpr T value = Result::value;
-};
-
-template <typename IntegerSequence>
-struct inclusive_scan_integer_sequence {
-  using value_type = typename IntegerSequence::value_type;
-  using helper     = inclusive_scan_integer_sequence_helper<
-      reverse_integer_sequence<IntegerSequence>,
-      std::integral_constant<value_type, 0>>;
-  using type                        = typename helper::type;
-  static constexpr value_type value = helper::value;
-};
+// We don't need to roll our own for these since we require C++14 now
+using std::index_sequence;
+using std::integer_sequence;
+using std::make_index_sequence;
```

Comment:
Did you consider tracking down usage and replace?

Reply:
Yes, but in case someone's currently using our version in `Impl` for whatever reason I figured we should do that in a separate commit. Also just trying to keep this diff minimal.

Reply:
Should we mark them as deprecated to encourage the transition?

Reply:
Actually what makes these special?  Why do we just yank utilities out in #3535 but keep them here?

Reply:
I would propose to just just remove since it's in `Impl` (or at the very least deprecate).